### DISCUSSION TITLE
Training tokens for sentiments

### DISCUSSION THREAD

**Discussion by balachander1964** [explosion/spaCy#10430] [Help: Coding & Implementations]
Hi, I request you to let me know how to train individual tokens for positivity / negativity. That is, how to train each token for sentiments. I look forward to your reply. Thank you. Bala

**Comment by polm** [comment]
spaCy doesn't have any feature for training per-token classification like that. You can get sentiment scores for individual tokens using the [example polar component](https://github.com/explosion/projects/tree/v3/pipelines/polar_component). Note that using token-based sentiment analysis is highly error prone due linguistic features like negation.

There is a per-token `sentiment` attribute, but it's never been used.

For sentiment analysis, we generally recommend treating it as text classification, or for span classification if you need something more fine grained.
